inflate.c

```diff
@@ -1002,21 +1062,26 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
      */
   inf_leave:
     RESTORE();
+    int windowup = 0;
     if (INFLATE_NEED_UPDATEWINDOW(strm) &&
             (state->wsize || (out != strm->avail_out && state->mode < BAD &&
-                 (state->mode < CHECK || flush != Z_FINISH)))) {
+                 (state->mode < CHECK)))) {
         if (updatewindow(strm, strm->next_out, out - strm->avail_out)) {
             state->mode = MEM;
             return Z_MEM_ERROR;
         }
+        windowup = 1;
     }
     in -= strm->avail_in;
     out -= strm->avail_out;
     strm->total_in += in;
     strm->total_out += out;
     state->total += out;
-    if (INFLATE_NEED_CHECKSUM(strm) && (state->wrap & 4) && out)
-        strm->adler = state->check = UPDATE(state->check, strm->next_out - out, out);
+
+    /* TODO: Figure out this thing's drama: */
```

Comment:
Yes this and `windowup` are odd..

Reply:
I actually suspect that with Z_FLUSH back in we don't need to check this anymore.  That hack _might_ have been to fix this scenario.  I'm almost afraid to yank it out, though.  At worst it's 2 extra checks.

Reply:
Better to yank it if we don't need it imo..

Reply:
I'll give it a shot, let's see if it passes CI.

Reply:
Looks to be passing.